### PR TITLE
Store Elevation Min & Max Per Study Area

### PR DESCRIPTION
**Pull request by waltaskew**
- As each chunk is processed into a feature matrix, update the meta
  store for the chunk's study area to maintain the min & max elevation
  per study area. We need this information to scale elevation values for
  the model.

### CODE REVIEW

**Review by rsutormin — CHANGES_REQUESTED** [review]

File: usl_pipeline/cloud_functions/main.py [hunk 1]
```diff
@@ -74,21 +84,54 @@ def _build_feature_matrix_from_archive(
...
+      fd: The file containing elevation data.
+
+    Returns:
+      A tuple of (array, metadata) for the elevation matrix and
+      metadata describing the feature matrix. The metadata dictionary
+      includes keys "elevation_min" and "elevation_max" for the min
+      and max values in the elevation matrix.
+    """
+    elevation = elevation_readers.read_from_geotiff(
+        rasterio.io.MemoryFile(fd.read())
```
[the remark this answers is not in the document]
Comment (waltaskew) [reply usl_pipeline/cloud_functions/main.py:108]: These are the chunks, so we should be fine.

File: usl_pipeline/cloud_functions/main.py [hunk 2]
```diff
@@ -74,21 +84,54 @@ def _build_feature_matrix_from_archive(
...
+    ).data
+
+    if elevation is None:
+        raise ValueError("Elevation file unexpectedly empty.")
+
+    metadata = {
```
[the remark this answers is not in the document]
Comment (waltaskew) [reply usl_pipeline/cloud_functions/main.py:114]: Can you make a more concrete suggestion for improvement?

[the remark this answers is not in the document]
Comment (rsutormin) [reply usl_pipeline/cloud_functions/main.py:114]: Yeah, I mean some structure with typed fields with verifiable field names. For now you can make a typo in a field key string (or if you expect to see something other than float as a value) and nothing would warn you except if you make some integration tests.

[the remark this answers is not in the document]
Comment (waltaskew) [reply usl_pipeline/cloud_functions/main.py]: I agree, this is better as a struct. Done.

**Review by rsutormin — APPROVED** [review]

**Pull request merged**